industrial_ci/src/tests/source_tests.sh

```diff
@@ -168,6 +168,12 @@ if [ "$CATKIN_LINT" == "true" ] || [ "$CATKIN_LINT" == "pedantic" ]; then
     ici_time_end  # catkin_lint
 fi
 
+if [ "$YAML_LINT" == "true" ]; then
```

Comment:
> - Concern:
>  - File format checking may not be a core of CI functionality. So I'm not sure if that goes into `source_tests.sh`, which is kind of core of industrial_ci IMO. But then I'm not sure either what should be the 'core' steps of CI.

"Core" steps for CI can be those that happen in a chain? E.g. `filling dependency -> build -> install (-> test)` (which is what `rosdep` and `Catkin` chain do).

I'm not sure, but things like file formatting check (which is what this PR adds) can be done independently from that chain, and thus could be run separately from the "main" sequence (source_tests.sh etc.) As there can be more convenient features like this that can run independently, I don't feel comfortable to keep inserting into the main sequence.

Moved back to WIP.